src/main/java/seedu/algobase/ui/action/UiParser.java

```diff
@@ -0,0 +1,15 @@
+package seedu.algobase.ui.action;
+
+import seedu.algobase.logic.parser.exceptions.ParseException;
+
+/**
+ * Represents a Parser that is able to parse user input into a {@code Command} of type {@code T}.
+ */
+public interface UiParser<T extends UiAction> {
```

Comment:
If `UiParser` parses user input into a `Command` of type `T` then here should be `<T extends Command>`? Probably the description is a bit off?

Reply:
I was using `Command` initially but switched. Good catch!